Rubberduck.Core/Settings/ToDoListSettings.cs

```diff
@@ -26,21 +28,83 @@ public ToDoMarker[] ToDoMarkers
             }
         }
 
+        public ToDoExplorerColumns ColumnHeaderInformation { get; set; }
+
         /// <Summary>
         /// Default constructor required for XML serialization.
         /// </Summary>
         public ToDoListSettings()
         {
         }
 
-        public ToDoListSettings(IEnumerable<ToDoMarker> defaultMarkers)
+        public ToDoListSettings(IEnumerable<ToDoMarker> defaultMarkers, ToDoExplorerColumns columnHeadings)
         {
             _markers = defaultMarkers;
+            ColumnHeaderInformation = columnHeadings;
         }
 
         public bool Equals(ToDoListSettings other)
         {
-            return other != null && ToDoMarkers.SequenceEqual(other.ToDoMarkers);
+            return other != null 
+                && ToDoMarkers.SequenceEqual(other.ToDoMarkers)
+                && ColumnHeaderInformation.Equals(other.ColumnHeaderInformation);
+        }
+    }
+
+    public class ToDoExplorerColumns : IEquatable<ToDoExplorerColumns>
+    {
+        public ToDoExplorerColumn DescriptionColumn { get; set; }
+        public ToDoExplorerColumn ProjectColumn { get; set; }
+        public ToDoExplorerColumn ModuleColumn { get; set; }
+        public ToDoExplorerColumn LineNumberColumn { get; set; }
```

Comment:
so if we wanted to make this more extensible and less prone to breakage, these could be changed into a Collection and update `ToDoExplorerColumn` itself to contain a `HeaderName` or something like that.

Reply:
As in `ObservableCollection<ToDoExplorerColumn> ColumnHeadersInformation`? That's what comes to mind when you suggest a collection and then implementing `INotifyPropertyChanged` when a column`s DisplayIndex is changed.